pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -202,14 +201,17 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 		return util.StatusAllOK
 	}
 
-	err = c.ensureFinalizer(typeConfig)
+	updated, err := c.ensureFinalizer(typeConfig)
 	if err != nil {
 		runtime.HandleError(errors.Wrapf(err, "Failed to ensure finalizer for FederatedTypeConfig %q", key))
 		return util.StatusError
+	} else if updated && util.IsNamespaceKind(typeConfig) {
```

Comment:
I'm assuming you are detecting the addition of the namespace ftc by noting when the finalizer is added.  I encourage you to document your intent so that future maintainers don't have to wonder about the following question:

Why is the successful addition of the finalizer to the namespace ftc a reason to reconcile namespaced FTCs?  Sync controller start is predicated on being about to source the namespace ftc from the cache, and does not depend on that ftc having a finalizer.

Reply:
Done.